### PR TITLE
Remove tests that test initialization (+ side effects)

### LINKED ISSUE #1960: refactor(?):  remove tests about #initialize setting attributes
From  [maint/tech debt storytime discussion Wed 10/19](https://docs.google.com/document/d/1vTv9ySljSgJ-zcrH2ZztDpdUtnVCxe_hgfZnIehkPGs/edit#heading=h.a6kqlslw7c5o):

As JLitt surfaced, tests like the below are not useful:

[spec/services/checksum_validator_spec.rb#L30-L44](https://github.com/sul-dlss/preservation_catalog/blob/main/spec/services/checksum_validator_spec.rb#L30-L44)

Look in all prescat spec files for similar.

### PR DESCRIPTION
**Pull request by mjgiarlo**
## Why was this change made? 🤔

Fixes #1960

This commit removes tests that are unnecessary as they test behavior that is basic Ruby behavior. This turned out to be a little tricky with the very tangly `CompleteMoabHandler`---its initialization tests covered two tiny private methods that I think we *do* want to continue testing. Instead of elevating those methods to the public API of the CMH class, or testing private methods, I decided to go the route of extracting those two methods to Rails helpers. While I generally am not in favor of leaning on helpers, I felt it was defensible over, say, creating one or two new service classes wrapped around the (small) methods.

## How was this change tested? 🤨

CI

### CODE REVIEW

File: spec/services/complete_moab_handler_spec.rb [hunk 1]
```diff
@@ -24,63 +24,6 @@
...
-        '-666' => '-666',
-        'vv001' => 'vv001',
-        'asdf' => 'asdf'
-      }.each do |k, v|
-        it "by parsing '#{k}' to '#{v}'" do
-          complete_moab_handler = described_class.new(druid, nil, k, ms_root)
-          expect(complete_moab_handler.incoming_size).to eq v
-        end
-      end
-    end
```
Comment (mjgiarlo) [inline spec/services/complete_moab_handler_spec.rb:71]: These tests moved.

File: spec/helpers/application_helper_spec.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        'v0006' => 'v0006',
+        '-666' => '-666',
+        'vv001' => 'vv001',
+        'asdf' => 'asdf'
+      }.each do |k, v|
+        it "by parsing '#{k}' to '#{v}'" do
+          expect(string_to_int(k)).to eq(v)
+        end
+      end
+    end
```
Comment (mjgiarlo) [inline spec/helpers/application_helper_spec.rb:41]: This is moved code.

File: app/services/complete_moab_handler.rb [hunk 3]
```diff
@@ -337,19 +337,4 @@ def version_comparison_results
...
-    return val[1..].to_i if val.instance_of?(String) && val.match(/^v\d+$/)
-    val
-  end
-
-  def string_to_int(val)
-    return if val.blank?
-    return val if val.instance_of?(Integer) # NOTE: negative integers caught with validation
-    return val.to_i if val.instance_of?(String) && val.scan(/\D/).empty?
-    val
-  end
```
Comment (mjgiarlo) [inline app/services/complete_moab_handler.rb:354]: This code moved into a helper.

File: app/helpers/application_helper.rb [hunk 4]
```diff
@@ -1,4 +1,18 @@
...
+    return val[1..].to_i if val.instance_of?(String) && val.match(/^v\d+$/)
+    val
+  end
+
+  def string_to_int(val)
+    return if val.blank?
+    return val if val.instance_of?(Integer) # NOTE: negative integers caught with validation
+    return val.to_i if val.instance_of?(String) && val.scan(/\D/).empty?
+    val
+  end
```
Comment (mjgiarlo) [inline app/helpers/application_helper.rb:17]: This is the moved code.

**Review by justinlittman — APPROVED** [review]

**Pull request merged**